### DISCUSSION TITLE
Nodes in relation geometries

### DISCUSSION THREAD

**Discussion by Helium314** [streetcomplete/StreetComplete#4338] [Q&A] [answered]
I noticed that  relation geometries created by `ElementGeometryCreator` only contain ways, but not nodes.
Why is this?

When changing it to include nodes, I found that nodes are not highlighted when relation geometry is highlighted, so including nodes changes (almost) nothing here.
But a difference is that when including nodes, `geometry.getBounds()` may return a larger (and actually more correct) bbox, which might be relevant in some cases.

**Comment by westnordost** [comment, marked as answer]
Because the element geometry is used for display in tangram-es and iirc tangram only knows points, lines and polygons.

**Comment by westnordost** [comment]
What do you need the bbox of relations for btw?

**Reply by Helium314** [reply]
Currently I don't think there is anything where it's really needed, because `ElementGeometry.getBounds()` is called rarely.

Where I actually noticed it is when testing my old cache version (where all elements are in tiles, not just nodes) same way as I do with the cache from PR.
Sometimes relations were not returned by the cache, but only from database. This is because that cache is using `getBounds` to check whether an element is inside a given bbox, but database gets relations via its nodes (directly and way nodes).

Some places where `getBounds` is used and theoretically elements could wrongly be missed:
`MapDataWithEditsSource.modifyBBoxMapData`: modified relation may wrongly not get updated
`OsmQuestController.createQuestsForElementDeferred`: the bbox for map data near a relation could be too small
`AddHousenumber.getApplicableElements`: building could wrongly be seen as having no address
`StyleableOverlayManager.updateStyledElements`: relation seen as off-screen is not updated (currently open PR)

Though in all cases I doubt that the incorrect bbox will lead to any real issues...

[Edit: Summarized: I would (naively) expect `ElementGeometry.getBounds()` to return the minimum bbox fully containing the element, and some uses of `getBounds()` in SC seem to expect it as well]
